qt/scientific_interfaces/Indirect/Reduction/ISISEnergyTransferPresenter.cpp

```diff
@@ -0,0 +1,257 @@
+
+  for (auto const &error : errors) {
+    if (!error.empty())
+      uiv.addErrorMessage(QString::fromStdString(error));
+  }
+
+  QString error = uiv.generateErrorMessage();
+  showMessageBox(error);
+
+  return validateInstrumentDetails() && uiv.isAllInputValid();
+}
+
+void IETPresenter::notifyRunClicked() { runTab(); }
+
+void IETPresenter::run() {
+  InstrumentData instrumentData = getInstrumentData();
+  IETRunData runData = m_view->getRunData();
+
+  connect(m_batchAlgoRunner, SIGNAL(batchComplete(bool)), this, SLOT(algorithmComplete(bool)));
+  disconnect(m_batchAlgoRunner, SIGNAL(batchComplete(bool)), this, SLOT(plotRawComplete(bool)));
+
+  m_outputGroupName = m_model->runIETAlgorithm(m_batchAlgoRunner, instrumentData, runData);
+}
+
+void IETPresenter::algorithmComplete(bool error) {
+  disconnect(m_batchAlgoRunner, SIGNAL(batchComplete(bool)), this, SLOT(algorithmComplete(bool)));
+
+  if (!error) {
+    m_outputWorkspaces = m_model->groupWorkspaces(m_outputGroupName, m_view->getGroupOutputOption());
+    m_pythonExportWsName = m_outputWorkspaces[0];
+
+    setOutputPlotOptionsWorkspaces(m_outputWorkspaces);
+    m_view->setSaveEnabled(true);
+  }
+}
+
+void IETPresenter::notifyPlotRawClicked() {
+  InstrumentData instrumentData = getInstrumentData();
+  IETPlotData plotParams = m_view->getPlotData();
+  std::vector<std::string> errors = m_model->validatePlotData(plotParams);
+
+  if (errors.empty()) {
+    m_view->setPlotTimeIsPlotting(true);
+
+    disconnect(m_batchAlgoRunner, SIGNAL(batchComplete(bool)), this, SLOT(algorithmComplete(bool)));
+    connect(m_batchAlgoRunner, SIGNAL(batchComplete(bool)), this, SLOT(plotRawComplete(bool)));
+
+    m_model->plotRawFile(m_batchAlgoRunner, instrumentData, plotParams);
+  } else {
+    m_view->setPlotTimeIsPlotting(false);
+    for (auto const &error : errors) {
+      if (!error.empty())
+        showMessageBox(QString::fromStdString(error));
+    }
+  }
+}
+
+void IETPresenter::plotRawComplete(bool error) {
+  disconnect(m_batchAlgoRunner, SIGNAL(batchComplete(bool)), this, SLOT(plotRawComplete(bool)));
+
+  if (!error) {
+    auto const filename = m_view->getFirstFilename();
+    std::filesystem::path fileInfo(filename);
+    auto const name = fileInfo.filename().string();
+    m_plotter->plotSpectra(name + "_grp", "0", IndirectSettingsHelper::externalPlotErrorBars());
+  }
+
+  m_view->setPlotTimeIsPlotting(false);
+}
+
+void IETPresenter::notifySaveClicked() {
+  IETSaveData saveData = m_view->getSaveData();
+  for (auto const &workspaceName : m_outputWorkspaces)
+    if (doesExistInADS(workspaceName))
+      m_model->saveWorkspace(workspaceName, saveData);
+}
+
+void IETPresenter::notifySaveCustomGroupingClicked() {
+  InstrumentData instrumentData = getInstrumentData();
+  std::string customGrouping = m_view->getCustomGrouping();
+
+  std::string customGroupingOutput = IETGroupingConstants::GROUPING_WS_NAME;
```

Comment:
why are we creating this intermediate variable?

Reply:
Just realised I'm obsessed with variables. As I said, it just makes the code looks more readable. But I will fix these as they are not necessary

Reply:
I usually agree with making things readable, but on this occasion since we do not edit the variable and the original name is clear its better to just use it directly.

Reply:
Okay